$NetBSD: patch-aa,v 1.1.1.1 1999/12/02 20:46:35 rh Exp $

Our perl_module is named mod_perl.so, not libperl.so
Get rid of annoying interaction, partly by commenting out some code,
partly by adding support for a few environment variables.

--- Makefile.PL.dist	Wed Nov 17 07:40:47 1999
+++ Makefile.PL	Thu Dec  2 19:38:08 1999
@@ -24,10 +24,10 @@
 %neededmodules =
     ( 
       'mod_perl.c' => { name	    => 'perl_module', 
-			path	    => ['$apache_src/modules/perl/libperl.so', '$EPBINDIR/modules/libperl.so'],
+			path	    => ['$apache_src/modules/perl/mod_perl.so', '$EPBINDIR/modules/mod_perl.so'],
 			win32path   => ['$EPBINDIR/modules/apachemoduleperl.dll'],
 			win32path   => ['$mpdll'],
-			file	    => 'libperl.so',
+			file	    => 'mod_perl.so',
 			win32file   => 'apachemoduleperl.dll',
 		      },
       'mod_dir.c'  => { name	    => 'dir_module', 
@@ -328,7 +328,8 @@
 
     if (-e "$apache_src/httpd.h" || -e "$apache_src/main/httpd.h" || -e "$apache_src/include/httpd.h")
 	    {
-	    $b = GetYesNo ("Use $apache_src as Apache source", 'y') if (!$b) ;
+	    # $b = GetYesNo ("Use $apache_src as Apache source", 'y') if (!$b) ;
+	    $b = 1;
 	    }
     }
 
@@ -519,7 +520,8 @@
             $EPGROUP = GetString ("Group to run httpd", $EPGROUP) ;
             }
         
-        $EPHTTPD = "$apache_src/httpd" ;
+	$EPHTTPD = $ENV{APACHE};
+        $EPHTTPD = "$apache_src/httpd" if (! -e $EPHTTPD);
         $EPHTTPD = "$apache_src/httpsd" if (!-e $EPHTTPD && -e "$apache_src/httpsd") ;
         
         if (!-e $EPHTTPD)
@@ -565,7 +567,9 @@
     $EPBINDIR	  = dirname ($EPHTTPD) ;    
     $EPMODPERL	  = '' ;
     
-    $addmodpath = '' ;
+    # $addmodpath = '' ;
+    $addmodpath = $ENV{APACHE_MODULES};
+    $addmodpath = '' if (! -d $addmodpath);
     while (($mod, $opt) = each %neededmodules)
 	{
 	if (!grep (/$mod/, @modules))
